test/packages/good/elasticsearch/transform/metadata_united/transform.json

```diff
@@ -0,0 +1,41 @@
+{
+    "source": {
+        "index": [
+            "metrics-endpoint.metadata_current_default*",
+            ".fleet-agents*"
+        ]
+    },
+    "dest": {
+        "index": ".metrics-endpoint.metadata_united_default"
+    },
+    "frequency": "1s",
+    "sync": {
+        "time": {
+            "delay": "4s",
+            "field": "updated_at"
+        }
+    },
+    "pivot": {
+        "aggs": {
+            "united": {
+                "scripted_metric": {
+                    "init_script": "state.docs = []",
+                    "map_script": "state.docs.add(new HashMap(params['_source']))",
+                    "combine_script": "return state.docs",
+                    "reduce_script": "def ret = new HashMap(); for (s in states) { for (d in s) { if (d.containsKey('Endpoint')) { ret.endpoint = d } else { ret.agent = d } }} return ret"
+                }
+            }
+        },
+        "group_by": {
+            "agent.id": {
+                "terms": {
+                    "field": "agent.id"
+                }
+            }
+        }
+    },
```

Comment:
As a follow up it'd be nice to add in `elastic-package` a way to test transforms.

Reply:
opened
- https://github.com/elastic/elastic-package/issues/859